src/managers/GuildEventManager.js

```diff
@@ -0,0 +1,171 @@
+    super(guild.client, GuildEvent, iterable);
+
+    /**
+     * The guild this manager belongs to
+     * @type {Guild}
+     */
+    this.guild = guild;
+  }
+
+  /**
+   * The cache of this manager
+   * @type {Collection<Snowflake, GuildEvent>}
+   * @name GuildEventManager#cache
+   */
+
+  /**
+   * Data that resolves to give a GuildEvent object. This can be:
+   * * A GuildEvent object
+   * * A Snowflake
+   * @typedef {GuildEvent|Snowflake} GuildEventResolvable
+   */
+
+  /**
+   * Options used to fetch a single guild event from a guild.
+   * @typedef {BaseFetchOptions} FetchGuildEventOptions
+   * @property {GuildEventResolvable} guildEvent The guild event to fetch
+   */
+
+  /**
+   * Options used to fetch multiple guild events from a guild.
+   * @typedef {Object} FetchGuildEventsOptions
+   * @property {boolean} [withUserCounts=true] Whether the number of users subscribed to the guild event
+   * should be returned
+   */
+
+  /**
+   * Obtains one or multiple guild events from Discord, or the guild cache if it's aready available.
+   * @param {GuildEventResolvable|FetchGuildEventOptions|FetchGuildEventsOptions} [options] The id of the guild event or
+   * options
+   * @returns {Promise<GuildEvent|Collection<Snowflake, GuildEvent>>}
+   */
+  async fetch(options = {}) {
+    const id = this.resolveId(options) ?? this.resolveId(options.guildEvent);
+
+    if (id) {
+      if (!options.force) {
+        const existing = this.cache.get(id);
+        if (existing) return existing;
+      }
+
+      const data = await this.client.api('guild-events', id).get();
+      return this._add(data, options.cache);
+    }
+
+    const data = await this.client.api
+      .guilds(this.guild.id)
+      .events.get({ query: { with_user_counts: options.withUserCounts ?? true } });
+    return data.reduce(
+      (coll, guildEvent) => coll.set(guildEvent.id, new GuildEvent(this.client, guildEvent)),
+      new Collection(),
+    );
+  }
+
+  /**
+   * Options used to create a guild event.
+   * @typedef {Object} GuildEventCreateOptions
+   * @property {string} name The name of the event
+   * @property {Date} scheduledStartTime The time to schedule the event at
+   * @property {PrivacyLevel|number} privacyLevel The privacy level of the event
+   * @property {GuildEventEntityType|number} entityType The scheduled entity type of the event
+   * @property {string} [description] The description of the event
+   * @property {StageChannel|Snowflake} [channel] The stage channel of the event
+   */
+
+  /**
+   * Creates a new guild event.
+   * @param {GuildEventCreateOptions} options Options for creating the guild event
+   * @returns {Promise<GuildEvent>}
+   */
+  async create(options) {
+    if (typeof options !== 'object') throw new TypeError('INVALID_TYPE', 'options', 'object', true);
+    let { privacyLevel, entityType, channel } = options;
+
+    if (!privacyLevel) throw new TypeError('INVALID_TYPE', 'privacyLevel', 'string or number', false);
```

Comment:
this would throw on `0`, granted 0 is not a valid privacy level but the error would seem to indicate that neither a string or a number was passed

Reply:
Not sure why I chose to validate `privacyLevel` here. The lib leaves the validation of these types of required values for the api. Changed it in f1e6d8e